### PR TITLE
[NavigationDrawer] Adding MDCElevation support

### LINKED ISSUE #8015: [NavigationDrawer] Internal issue: b/138204931
This was filed as an internal issue. If you are a Googler, please visit [b/138204931](http://b/138204931) for more details.

---
#### Internal data
- Associated internal bug: [b/138204931](http://b/138204931)

### PR DESCRIPTION
**Pull request by yarneo**
This adds elevation conformance and support to the MDCBottomDrawerViewController.

Closes #8015

### CODE REVIEW

File: components/NavigationDrawer/src/MDCBottomDrawerViewController.h [hunk 1]
```diff
@@ -106,6 +109,17 @@
...
+/**
+ This block is called after a change of the bottom drawer view controller's elevation or one of its
+ view hierarchy ancestors. Use this block to respond to elevation changes in the view or its
+ ancestor views.
+ @param elevation The @c mdc_currentElevation plus the @c mdc_currentElevation of all ancestor
+ views.
+ @param object This bottom drawer view controller.
+ */
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)
+    (MDCBottomDrawerViewController *_Nonnull object, CGFloat elevation);
```
Comment (bryanoltman) [inline components/NavigationDrawer/src/MDCBottomDrawerViewController.h]: `object` should be changed to `bottomDrawerViewController`

File: components/NavigationDrawer/src/MDCBottomDrawerViewController.m [hunk 2]
```diff
@@ -191,10 +195,19 @@ - (void)setElevation:(MDCShadowElevation)elevation {
   if ([self.presentationController isKindOfClass:[MDCBottomDrawerPresentationController class]]) {
     MDCBottomDrawerPresentationController *bottomDrawerPresentationController =
         (MDCBottomDrawerPresentationController *)self.presentationController;
+    BOOL elevationDidChange =
+        !MDCCGFloatEqual(bottomDrawerPresentationController.elevation, elevation);
```
Comment (bryanoltman) [inline components/NavigationDrawer/src/MDCBottomDrawerViewController.m:199]: It's probably cleaner to put the following elevation assignment and call to `mdc_elevationDidChange` inside of `if (!MDCCGFloatEqual(bottomDrawerPresentationController.elevation, elevation)` – we don't need to assign if the value hasn't changed.

**Review by bryanoltman — APPROVED** [review]

File: components/NavigationDrawer/src/MDCBottomDrawerViewController.h (hunk 1 above)
Comment (yarneo) [reply components/NavigationDrawer/src/MDCBottomDrawerViewController.h]: done.

File: components/NavigationDrawer/src/MDCBottomDrawerViewController.m (hunk 2 above)
Comment (yarneo) [reply components/NavigationDrawer/src/MDCBottomDrawerViewController.m:199]: done.

**Pull request merged**